cookiecutter_django_ninja_example/users/api/views.py

```diff
@@ -1,26 +1,58 @@
-from rest_framework import status
-from rest_framework.decorators import action
-from rest_framework.mixins import ListModelMixin
-from rest_framework.mixins import RetrieveModelMixin
-from rest_framework.mixins import UpdateModelMixin
-from rest_framework.response import Response
-from rest_framework.viewsets import GenericViewSet
+from ninja import Router
+from ninja.errors import HttpError
 
 from cookiecutter_django_ninja_example.users.models import User
 
-from .serializers import UserSerializer
+from .schemas import UserSchema
+from .schemas import UserUpdateSchema
 
+users_router = Router(tags=["users"])
 
-class UserViewSet(RetrieveModelMixin, ListModelMixin, UpdateModelMixin, GenericViewSet):
-    serializer_class = UserSerializer
-    queryset = User.objects.all()
-    lookup_field = "username"
 
-    def get_queryset(self, *args, **kwargs):
-        assert isinstance(self.request.user.id, int)
-        return self.queryset.filter(id=self.request.user.id)
+# error handling in django-ninja is different from DRF
+# see https://django-ninja.dev/guides/errors/
+# do we need to add a custom error handler function in template?
 
-    @action(detail=False)
-    def me(self, request):
-        serializer = UserSerializer(request.user, context={"request": request})
-        return Response(status=status.HTTP_200_OK, data=serializer.data)
+
+@users_router.get("me/", response=UserSchema, url_name="user-me")
+def user_me(request):
+    return request.user
+
+
+@users_router.get("/", response=list[UserSchema], url_name="user-list")
+def user_list(request):
+    return User.objects.all()
+
+
+@users_router.get("{username}/", response=UserSchema, url_name="user-detail")
+def user_detail(request, username: str):
+    try:
+        return User.objects.get(username=username)
+    except User.DoesNotExist as exc:
+        raise HttpError(404, "User not found") from exc
```

Comment:
Can be simplified with `get_object_or_404(User, username=username)`

```suggestion
    return get_object_or_404(User, username=username)
```